src/lxc/tools/lxc_top.c

```diff
@@ -35,16 +34,24 @@ struct blkio_stats {
 	uint64_t total;
 };
 
-struct stats {
-	uint64_t mem_used;
-	uint64_t mem_limit;
-	uint64_t memsw_used;
-	uint64_t memsw_limit;
+struct cpu_stats {
+	uint64_t use_nanos;
+	uint64_t use_user;
+	uint64_t use_sys;
+};
+
+struct mem_stats {
+	uint64_t used;
+	uint64_t limit;
+	uint64_t swap_used;
```

Comment:
it's not a swap, really. It's mem+swap. In cgroupv1 we have a combined page counter for memory+swap:
https://github.com/torvalds/linux/blob/1722389b0d863056d78287a120a1d6cadb8d4f7b/mm/memcontrol.c#L1536

for cgroup-v2, yes, we have a separate `swap` page counter.

I would go the same way, as it's done in the kernel:
https://github.com/torvalds/linux/blob/1722389b0d863056d78287a120a1d6cadb8d4f7b/include/linux/memcontrol.h#L191

and use union:
```
	union {
                struct {
                        uint64_t swap_used;
	                uint64_t swap_limit;
                }; /* v2 only */
                struct {
                        uint64_t memsw_used;
	                uint64_t memsw_limit;
                }; /* v1 only */
	};
```

it makes it a bit cleaner.

Reply:
In cgroup2, should we keep the `memsw_used_bytes` column in `lxc-top -b` and calculate that value by adding `memory.current + memory.swap.current`?